addons/l10n_it_edi/models/res_company.py

```diff
@@ -107,3 +110,36 @@ def _check_tax_representative(self):
                 raise ValidationError(_("Your tax representative partner must have a tax number."))
             if not record.l10n_it_tax_representative_partner_id.country_id:
                 raise ValidationError(_("Your tax representative partner must have a country."))
+
+    @api.onchange('l10n_it_edi_mode')
+    def _onchange_l10n_it_edi_mode(self):
```

Comment:
The thing with onchanges is that they are triggered not just by a `write` but also by modifications in the UI (if I am not mistaken). I don't think it's intended for them to make modifications in the DB, especially not far-reaching ones like this?

Personally, I would either just provide a method that can be called by the res.config.settings when changing the active proxy user, or override the `write()` method.